pkg/apis/apps/v1beta2/defaults.go

```diff
@@ -154,17 +130,8 @@ func SetDefaults_Deployment(obj *appsv1beta2.Deployment) {
 
 func SetDefaults_ReplicaSet(obj *appsv1beta2.ReplicaSet) {
 	labels := obj.Spec.Template.Labels
-
-	// TODO: support templates defined elsewhere when we support them in the API
-	if labels != nil {
-		if obj.Spec.Selector == nil {
-			obj.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: labels,
-			}
-		}
-		if len(obj.Labels) == 0 {
-			obj.Labels = labels
-		}
+	if labels != nil && len(obj.Labels) == 0 {
```

Comment:
Delete this. You are setting the labels of the ReplicaSet  to the labels of the PodTemplateSpec. You do not want to do this.